packages/grid/x-data-grid/src/components/virtualization/GridVirtualScroller.tsx

```diff
@@ -77,10 +77,11 @@ function GridVirtualScroller(props: GridVirtualScrollerProps) {
       <Scroller className={classes.scroller} {...getScrollerProps()} ownerState={rootProps}>
         <TopContainer>
           <GridHeaders />
-          <GridOverlays />
           <rootProps.slots.pinnedRows position="top" virtualScroller={virtualScroller} />
         </TopContainer>
 
+        <GridOverlays />
```

Comment:
@romgrk 

Along with `top: 'var(--DataGrid-headersTotalHeight)',` fixes this issue:


https://github.com/mui/mui-x/assets/13808724/1e95e102-acb9-444d-87bd-322d3a68d449